bundles/de.uka.ipd.sdq.simucomframework/src/de/uka/ipd/sdq/simucomframework/resources/IAssemblyAllocationLookup.java

```diff
@@ -0,0 +1,59 @@
+package de.uka.ipd.sdq.simucomframework.resources;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.palladiosimulator.pcm.core.composition.AssemblyContext;
+
+/**
+ * Through the IAssemblyAllocationLookup interface it is possible to access the
+ * current allocations of assembly contexts to resource containers.
+ * 
+ * The interface has been introduced to account for SimuLizar's dynamic nature
+ * with allocations being created or changed during simulation time.
+ * 
+ * Furthermore, this interfaces allows for multiple contexts to share the same
+ * lookup information.
+ * 
+ * @author Sebastian Krach
+ *
+ */
+public interface IAssemblyAllocationLookup<AllocationType> {
+
+    /**
+     * Provides a simple implementation of the interface based on HashMap. The
+     * HashMap itself needs to be kept externally in order to provide editing
+     * support.
+     */
+    public class HashMapAssemblyAllocationLookup<AllocationType> implements IAssemblyAllocationLookup<AllocationType> {
+        protected final Map<String, AllocationType> internalMap;
+
+        public HashMapAssemblyAllocationLookup(Map<String, AllocationType> delegateMap) {
+            internalMap = Collections.unmodifiableMap(delegateMap);
+        }
+
+        @Override
+        public AllocationType getAllocatedEntity(String assemblyContextId) {
+            return internalMap.get(assemblyContextId);
+        }
+    }
```

Comment:
I would prefer moving this into an independent type.

Reply:
Done.